be/src/exec/vectorized/arrow_to_starrocks_converter.cpp

```diff
@@ -580,7 +580,14 @@ struct ArrowConverter<AT, PT, is_nullable, is_strict, DateOrDateTimeATGuard<AT>,
                                    [[maybe_unused]] int divisor) {
         for (int i = 0; i < num_elements; ++i) {
             if constexpr (is_nullable) {
-                if (null_data[i] == DATUM_NULL) continue;
+                if (null_data[i] == DATUM_NULL) {
+                    if constexpr (pt_is_date<PT>) {
+                        data[i] = DefaultValueGenerator<DateValue>::next_value();
```

Comment:
if the data is null, i think the real data value could be any value, why do we need to set it to default value explicitly.

Reply:
Yep.  This behavior is a bit strange.  Maybe others can meet the problem again latter.

Add some comments or explanations?

Reply:
for nullable column, aggregation will first compare null flag, then compare the data
```c++
    Status do_visit(const vectorized::NullableColumn& column) {
        // TODO: maybe could skip compare rows that contains null
        // 1. Compare the null flag
        // 2. Compare the value if both are not null. Since value for null is just default value,
        //    which are equal, so just compare the value directly
        const NullData& null_data = column.immutable_null_column_data();
        for (size_t i = 1; i < column.size(); i++) {
            (*_tie)[i] &= (null_data[i - 1] == null_data[i]);
        }

        build_tie_for_column(column.data_column(), _tie);
        return Status::OK();
    }

    template <typename T>
    Status do_visit(const vectorized::BinaryColumnBase<T>& column) {
        auto& data = column.get_data();
        for (size_t i = 1; i < column.size(); i++) {
            (*_tie)[i] &= SorterComparator<Slice>::compare(data[i - 1], data[i]) == 0;
        }
        return Status::OK();
    }
```

Reply:
For broker/stream load, null date/datetime'sdefault value is set to DefaultValueGenerator<DateValue/TimestampValue>

Reply:
i will add some comments here

Reply:
> for nullable column, aggregation will first compare null flag, then compare the data
> 
> ```c++
>     Status do_visit(const vectorized::NullableColumn& column) {
>         // TODO: maybe could skip compare rows that contains null
>         // 1. Compare the null flag
>         // 2. Compare the value if both are not null. Since value for null is just default value,
>         //    which are equal, so just compare the value directly
>         const NullData& null_data = column.immutable_null_column_data();
>         for (size_t i = 1; i < column.size(); i++) {
>             (*_tie)[i] &= (null_data[i - 1] == null_data[i]);
>         }
> 
>         build_tie_for_column(column.data_column(), _tie);
>         return Status::OK();
>     }
> 
>     template <typename T>
>     Status do_visit(const vectorized::BinaryColumnBase<T>& column) {
>         auto& data = column.get_data();
>         for (size_t i = 1; i < column.size(); i++) {
>             (*_tie)[i] &= SorterComparator<Slice>::compare(data[i - 1], data[i]) == 0;
>         }
>         return Status::OK();
>     }
> ```

I think the aggregation comparison logic isn't reasonable, we shouldn't relay on data column content is default value if which is null.

Reply:
i am not sure if there exist other place use the null value's data

Reply:
@kangkaisen maybe we can modify both?
1. let the data for null value to be consistent
2. let the aggregation not check the data for null value?

Reply:
OK